src/core/jpg.js

```diff
@@ -839,6 +862,13 @@ var JpegImage = (function JpegImageClosure() {
             break;
 
           case 0xFFDA: // SOS (Start of Scan)
+            numScanSegments++;
```

Comment:
Do you really need to count? Can't you use a boolean `hasSeenSOS` ;)?

Reply:
Perhaps not, but I do find the current state slightly clearer w.r.t. the specification.
With the proposed change you'd need to compare `hasSeenSOS === false` below (before setting `hasSeenSOS = true;`), which seems slightly strange when we've in fact actually encountered a SOS marker by entering this block.